Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,230 @@
+# Regular Expression Literals
+
+- Authors: Hamish Knight, Michael Ilseman
+
+## Introduction
+
+We propose to introduce a first-class regular expression literal into the language that can take advantage of library support to offer extensible, powerful, and familiar textual pattern matching.
+
+This is a component of a larger string processing picture. We would like to start a focused discussion surrounding our approach to the literal itself, while acknowledging that evaluating the utility of the literal will ultimately depend on the whole picture (e.g. supporting API). To aid this focused discussion, details such as the representation of captures in the type system, semantic details, extensions to lexing/parsing, additional API, etc., are out of scope of this pitch and thread. Feel free to continue discussion of anything related in the [overview thread][overview].
+
+## Motivation
+
+Regular expressions are a ubiquitous, familiar, and concise syntax for matching text that satisfies a particular format, and extracting subranges from the match that contain interesting information. Syntactically, a regex literal in Swift should:
+
+- Leverage broader familiarity
+- Fit in with Swift's library extensibility story for literals (e.g. [`ExpressibleByStringInterpolation`][stringinterpolation])
+- Address source-destabilizing changes such as the parsing of delimiters
```

Comment:
I was confused by this bullet until I read on further, I could tell if it was talking about how Swift regular expression should support the parsing of delimiters (like matching parenthesis or quotation marks), but this is about the source-breaking implications of using `/` as the delimiter.

I think we can just delete this bullet since using the `/` isn't a motivation on its own—choosing `/` follows from the the underling motivation for Swift regex literals to be familiar.